main.cpp

```diff
@@ -40,29 +40,337 @@ Purpose:  This project continues developing Project3.
+{
+
+}
+
+void Restaurant::tableAmount(int platePerFamily, int families)
+{
+    while (families < platesServed) 
+    {
+        amountPerTable += (platePerFamily * amFoodPerFamily);
+        std::cout << "The Amount Of food served Total Is: " << amountPerTable << std::endl;
+        ++families;
+    }
+    std::cout << std::endl;
+}
+
+void Restaurant::Server::serveCustomer()
+{
+
+}
+
+void Restaurant::Server::cleanTable()
+{
+
+}
+
+float Restaurant::Server::countTip(float tip)
+{
+    return tip;
+}
 
 /*
  copied UDT 3:
  */
+struct Speakers
+{
+    std::string brand {"Vagnavox"};
+    int modelNum {214375};
+    int size = 12;
+    int time;
+    float freqRange = 15000.00f;
+    float impedence = 59.45f;
+    
+    Speakers();
+    ~Speakers()
+    {
+        std::cout << "Destroying Speaker Constructor!" << std::endl;
+    }
+    
+    void produceAudio();
+    void receiveSignal();
+    void vibrateCone();
+};
 
+Speakers::Speakers():
+size(),
+freqRange()
+{
+    std::cout << "Speakers Class Created" << std::endl;
+}
+
+void Speakers::produceAudio()
+{
+    while (time < 15)
+    {
+        std::cout << "Producing Audio!!!!!!!" << std::endl;
+        ++time;
+    }
+    std::cout << std::endl;
+}
+
+void Speakers::receiveSignal()
+{
+    std::cout << "Model Number: " << modelNum << std::endl;
+}
+
+void Speakers::vibrateCone()
+{
+    
+}
 /*
  new UDT 4:
  with 2 member functions
  */
+struct Equipment
+{
+    Laundromat machine;
+    Laundromat::WashingMachine pipe;
+    Restaurant dishWasher;
+    Restaurant::Server carrier;
+    Speakers tweaters;
+
+    Equipment();
+    ~Equipment()
+    {
+        std::cout << "Equipment Class Destroyed!" << std::endl;
+    }
 
+    void laundromatEquipment();
+    void restaurantEquipment();
+    void speakerEquipment();
+};
+Equipment::Equipment():
+machine(20, 34),
+pipe(24.3f, 43.1f, 9),
+dishWasher(),
+carrier(),
+tweaters()
+{
+    
+    std::cout << "Equipment Class Created " << std::endl;
+}
+
+void Equipment::laundromatEquipment()
+{
+    std::cout << "Laundromat Machines in Inventory: " << machine.numWashers << std::endl;
+    std::cout << "Laundromat Dryers in Inventory: " << machine.numDryers << std::endl;
+}
+
+void Equipment::restaurantEquipment()
+{
+     std::cout << "Restaurant Workers: " << dishWasher.workers << std::endl;
+    std::cout << "Amount of Server Carriers In Inventory: " << carrier.amOfPeopleServed << std::endl;
+}
+
+void Equipment::speakerEquipment()
+{
+    std::cout << "Speaker Size: " << tweaters.size << std::endl;
+    std::cout << "Speaker Frequency Range: " << tweaters.freqRange << std::endl;
+}
 /*
  new UDT 5:
  with 2 member functions
  */
+struct Data
+{
+    Laundromat equipment;
+    Restaurant personal;
+    Speakers parts;
+    Laundromat::WashingMachine components;
+
+    Data();
+    ~Data()
+    {
+        std::cout << "Data Class Destroyed!" << std::endl;
+    }
+    
+    void laundromatData();
+    void restaurantData();
```

Comment:
Name things what they are or what they do.
These are functions so the names need to describe what they do.
These function names do not describe what these functions do.
- [ ] revise